### PR TITLE
Add zsh completion

### PR DESCRIPTION
**Pull request by Vexu**

### CODE REVIEW

File: completions/_zig [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  "1: :{_describe 'command' __zig_commands}" \
+  "*:: :->args"
+
+case $state in
+  args)
+
+  case $words[1] in
+    build-exe | build-lib | build-obj | run | translate-c)
+      _arguments $__zig_general_options $__zig_compile_options $__zig_link_options $__zig_debug_options
+        ;;
```
Comment (daurnimator) [inline completions/_zig:188]: indentation here doesn't match the other cases.

**Review by daurnimator — APPROVED** [review]

**Comment by FireFox317** [comment]
Wouldn't it be better to auto-generate these bindings?

**Comment by Vexu** [comment]
> Wouldn't it be better to auto-generate these bindings?

Hardly seems worth the effort considering it takes less than a minute to just update this manually, but I could add a test to check if any of the `usage_*` texts in main have changed.

**Comment by data-man** [comment]
> Wouldn't it be better to auto-generate these bindings?

And in `ziglang/shell-completions` repo? It's will not trigger testing.

**Comment by andrewrk** [comment]
> And in `ziglang/shell-completions` repo? It's will not trigger testing.

I'd like to second this request @Vexu - mind maintaining this in a separate repo?

**Comment by Vexu** [comment]
https://github.com/ziglang/shell-completions

**Comment by daurnimator** [comment]
It would be much better to keep the completions in this repo with the same release cycle.
Otherwise as we add/remove flags things are likely to get out of sync; and its not going to be obvious which completion files to use with each release.

**Pull request closed without merging**